### PR TITLE
LPS-112298 Update Liferay.Util.selectEntity utility

### PR DESCRIPTION
**Pull request by markocikos**
Hey @wincent and @jbalsas,

This is a followup on https://github.com/wincent/liferay-portal/pull/262

Following the awesome [clarification](https://github.com/wincent/liferay-portal/pull/262#issuecomment-627491152), I rewrote utility to continue supporting custom event based contract with selectors view:
- I added the `selectEventName` prop. If `selectEventName` and `onSelect` are defined, we define a global listener for that event, that executes `onSelect` callback. There is no longer a delegation on CSS class.
- Usages continue to require `Liferay.Util.selectEntityHandler`, I reverted removal from previous PR. We can update `selectEntityHandler` util to modern patterns in a separate task.
- A minor tweak was needed on `selectEntityHandler`, as [this line](https://github.com/wincent/liferay-portal/commit/871eff1ab76b550ba3005b22728bc8cb7e077587#diff-3d16b13f5d58217f2a53ac02b9855e77L1339) is specific to AUI modal implementation.
- We now need to set `selectEventName` in usages.
- This approach makes `onSelect` more generic, fitting to its name. It can be used with a non-iframe modal, unlike in previous solution.
- I removed the test. Custom event handling in jest seems difficult to achieve, and I'm not completely sure that the approach in this PR is correct.

Please review the code.

Thank you!

### CODE REVIEW

**Comment by wincent** [comment]
Thanks @markocikos for the great description as always. I'm going to start CI running on this and look at the actual code in the morning.

**Comment by wincent** [comment]
ci:test:sf

**Comment by wincent** [comment]
ci:test:relevant

**Comment by wincent** [comment]
ci:test:bundle

**Comment by wincent** [comment]
> look at the actual code in the morning.

Ok, I lied. Seeing there is not much code I looked at just now. Superficially, the code looks fine to me. Would appreciate @jbalsas's opinion though on an architectural level, seeing as he's the one who came up with the original pattern.

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js [hunk 1]
```diff
@@ -204,6 +220,7 @@ const Modal = ({
 								{loading && <ClayLoadingIndicator />}
 								<Iframe
 									iframeBodyCssClass={iframeBodyCssClass}
+									onSelect={onSelect}
```
Comment (jbalsas) [inline modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js:223]: Why are we passing `onSelect` to the iframe? Is this meant to do something?

**Comment by jbalsas** [comment]
> I rewrote utility to continue supporting custom event based contract with selectors view

Nice!! Changes also seem minimal which is usually less scary :)

> Usages continue to require Liferay.Util.selectEntityHandler, I reverted removal from previous PR. We can update selectEntityHandler util to modern patterns in a separate task.

Sounds fair enough!

LGTM! 👍

**Comment by jbalsas** [comment]
> Would appreciate @jbalsas's opinion though on an architectural level, seeing as he's the one who came up with the original pattern.

There's no significant change here on arch level that I could appreciate... as we get into this a bit better, it might be worth revisit the idea with some fresh eyes. We've discussed the idea of a "remote widget" as an iframe+facilities... which might also fit in this scenario. Rather than having an event-based communication mechanism, it might be interesting to explore a more Reactish approach like injecting a `selectItem()` kind of callback that views would call when ready, thus removing the need for `eventName`... would be interesting to see how this could play out and how to make a smooth transition from where we are...

This feels like a good first step though!

File: modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js (hunk 1 above)
Comment (markocikos) [reply modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/modal/Modal.js:223]: Oops, that's leftover code from the old solution. I'll clean it up and resend.

**Comment by markocikos** [comment]
See https://github.com/wincent/liferay-portal/pull/265

**Pull request closed without merging**